src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -193,7 +193,19 @@ type NodeCode private () =
         }
 
     static member Parallel(computations: NodeCode<'T> seq) =
-        computations |> Seq.map (fun (Node x) -> x) |> Async.Parallel |> Node
+        let diagnosticsLogger = DiagnosticsThreadStatics.DiagnosticsLogger
+        let phase = DiagnosticsThreadStatics.BuildPhase
+
+        computations
+        |> Seq.map (fun (Node x) ->
+            async {
+                DiagnosticsThreadStatics.DiagnosticsLogger <- diagnosticsLogger
```

Comment:
This is not safe, as the logger we're passing is most probably not thread-safe. So we're betting on the computation to actually never use it.